crates/plugins/src/manager.rs

```diff
@@ -194,6 +194,61 @@ impl PluginManager {
         };
         Ok(plugin_manifest)
     }
+
+    pub async fn update_lock(&self) -> PluginManagerUpdateLock {
```

Comment:
I would expect this to return a `Result<PluginManagerUpdateLock>`. This would mean we wouldn't have to thread through the failure case to the lock guard which would only have two states: `Acquired` and `Denied`.

Do we have the three states because we explicitly want to ignore failed locks? That doesn't seem right...

Reply:
Yes.  The intention is to ignore failed locks, and fall back to the previous behaviour (as Lann described).  I'm happy to discuss whether this is the desired behaviour, but I wanted to avoid the situation of "if something goes wrong in the locking system then you can never update."